### PR TITLE
Add qa model and new settings in ml-commons

### PR DESCRIPTION
**Pull request by rbhavna**
### Description
add qa model and new settings documentation in ml-commons

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6744
Closes https://github.com/opensearch-project/documentation-website/issues/6752

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/cluster-settings.md [hunk 1]
```diff
@@ -254,6 +281,21 @@ plugins.ml_commons.allow_custom_deployment_plan: false
...
+
+### Setting
+
+```
+plugins.ml_commons.model_auto_deploy.enable: false
+```
+
+### Values
+
+- Default value: true
```
Comment (natebower) [inline _ml-commons-plugin/cluster-settings.md:296]:
```suggestion
- Default value: `true`
```

File: _ml-commons-plugin/cluster-settings.md [hunk 2]
```diff
@@ -329,7 +371,107 @@ plugins.ml_commons.connector_access_control_enabled: true
 - Default value: false
```
Comment (natebower) [inline _ml-commons-plugin/cluster-settings.md:371]:
```suggestion
- Default value: `false`
```

File: _ml-commons-plugin/cluster-settings.md [hunk 3]
```diff
@@ -329,7 +371,107 @@ plugins.ml_commons.connector_access_control_enabled: true
...
+```
+plugins.ml_commons.local_model.enabled: true
+```
 
+### Values
 
+- Default value: `true`
+- Valid values: `false`, `true`
+
+## Remote model eligible node roles
```
Comment (natebower) [inline _ml-commons-plugin/cluster-settings.md:389]: "Eligible node roles for remote models"?

File: _ml-commons-plugin/cluster-settings.md [hunk 4]
```diff
@@ -329,7 +371,107 @@ plugins.ml_commons.connector_access_control_enabled: true
...
+```
+
+### Values
+
+- Default value: `["data", "ml"]`, which allows externally hosted models to run on data nodes and ML nodes.
+
+
+## Local model eligible node roles setting
+
+This setting allows a cluster admin to control the types of nodes on which local models can run. The`plugins.ml_commons.only_run_on_ml_node` setting specifies for the model to only run on ML nodes. For a local model, if `plugins.ml_commons.only_run_on_ml_node` is set to `true`, the model will always run on ML nodes. If `plugins.ml_commons.only_run_on_ml_node` is set to `false`, the model will run on nodes defined in the `plugins.ml_commons.task_dispatcher.eligible_node_role.local_model` setting.
```
Comment (natebower) [inline _ml-commons-plugin/cluster-settings.md:406]:
```suggestion
This setting allows a cluster admin to control the types of nodes on which local models can run. The `plugins.ml_commons.only_run_on_ml_node` setting only allows the model to run on ML nodes. For a local model, if `plugins.ml_commons.only_run_on_ml_node` is set to `true`, then the model will always run on ML nodes. If `plugins.ml_commons.only_run_on_ml_node` is set to `false`, then the model will run on nodes defined in the `plugins.ml_commons.task_dispatcher.eligible_node_role.local_model` setting.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 5]
```diff
@@ -329,7 +371,107 @@ plugins.ml_commons.connector_access_control_enabled: true
...
+```
+plugins.ml_commons.task_dispatcher.eligible_node_role.remote_model: ["ml"]
+```
+
+### Values
+
+- Default value: `["data", "ml"]`, which allows externally hosted models to run on data nodes and ML nodes.
+
+
+## Local model eligible node roles setting
```
Comment (natebower) [inline _ml-commons-plugin/cluster-settings.md:404]: "Eligible node roles setting for local models"?

File: _ml-commons-plugin/cluster-settings.md [hunk 6]
```diff
@@ -329,7 +371,107 @@ plugins.ml_commons.connector_access_control_enabled: true
...
+plugins.ml_commons.task_dispatcher.eligible_node_role.remote_model: ["ml"]
+```
+
+### Values
+
+- Default value: `["data", "ml"]`
+
+## Enable remote inference
+
+This setting allows a cluster admin to enable remote inference on the cluster. If this setting is `false`, users will not be able to run register, deploy, or predict operations on any externally hosted model or create any connector for remote inference.
```
Comment (natebower) [inline _ml-commons-plugin/cluster-settings.md:420]:
```suggestion
This setting allows a cluster admin to enable remote inference on the cluster. If this setting is `false`, users will not be able to run register, deploy, or predict operations on any externally hosted model or create a connector for remote inference.
```

File: _ml-commons-plugin/cluster-settings.md [hunk 7]
```diff
@@ -329,7 +371,107 @@ plugins.ml_commons.connector_access_control_enabled: true
...
+```
+
+### Values
+
+- Default value: `true`
+- Valid values: `false`, `true`
+
+## Enable agent framework
+
+When set to `true`, this setting enables the agent framework (incuding agents and tools) on the cluster and allows users to run register, execute, delete, get, and search operations on an agent.
```
Comment (natebower) [inline _ml-commons-plugin/cluster-settings.md:435]:
```suggestion
When set to `true`, this setting enables the agent framework (including agents and tools) on the cluster and allows users to run register, execute, delete, get, and search operations on an agent.
```

File: _ml-commons-plugin/custom-local-models.md [hunk 8]
```diff
@@ -20,12 +20,14 @@ As of OpenSearch 2.11, OpenSearch supports local sparse encoding models.
...
 
+As of OpenSearch 2.13, OpenSearch supports local question answering models.
+
 Running local models on the CentOS 7 operating system is not supported. Moreover, not all local models can run on all hardware and operating systems.
 {: .important}
 
 ## Preparing a model
 
-For both text embedding and sparse encoding models, you must provide a tokenizer JSON file within the model zip file.
+For all the models, you must provide a tokenizer JSON file within the model zip file.
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:30]: ZIP file?

File: _ml-commons-plugin/custom-local-models.md [hunk 9]
```diff
@@ -321,3 +323,60 @@ The response contains the tokens and weights:
 ## Step 5: Use the model for search
 
 To learn how to use the model for vector search, see [Using an ML model for neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/#using-an-ml-model-for-neural-search).
+
+## Question Answering models
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:327]:
```suggestion
## Question answering models
```

File: _ml-commons-plugin/custom-local-models.md [hunk 10]
```diff
@@ -321,3 +323,60 @@ The response contains the tokens and weights:
...
+
+Question Answering model extracts the answer to the question from a given context. Currently ml-commons supports context in `text` format.
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:329]:
```suggestion
A question answering model extracts the answer to a question from a given context. ML Commons supports context in `text` format.
```

File: _ml-commons-plugin/custom-local-models.md [hunk 11]
```diff
@@ -321,3 +323,60 @@ The response contains the tokens and weights:
...
+{% include copy-curl.html %}
+
+Then send a request to deploy the model:
+
+```json
+POST _plugins/_ml/models/<model_id>/_deploy
+```
+{% include copy-curl.html %}
+
+To test a question answering, send the following request. It requires a `question` and the relevant `context` from which answer will be generated:
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:359]:
```suggestion
To test a question answering model, send the following request. It requires a `question` and the relevant `context` from which the answer will be generated:
```

File: _ml-commons-plugin/custom-local-models.md [hunk 12]
```diff
@@ -321,3 +323,60 @@ The response contains the tokens and weights:
...
+```json
+POST /_plugins/_ml/_predict/question_answering/<model_id>
+{
+  "question": "Where do I live?"
+  "context": "My name is John. I live in New York"
+}
+```
+{% include copy-curl.html %}
+
+The response gives the answer based on the context provided:
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:370]:
```suggestion
The response provides the answer based on the context:
```

**Review by natebower — COMMENTED** [review]
@rbhavna @kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _ml-commons-plugin/cluster-settings.md [hunk 13]
```diff
@@ -326,10 +368,110 @@ plugins.ml_commons.connector_access_control_enabled: true
...
+```
+plugins.ml_commons.local_model.enabled: true
+```
 
+### Values
+
+- Default value: `true`
+- Valid values: `false`, `true`
 
+## Remote model eligible node roles
```
Comment (kolchfa-aws) [inline _ml-commons-plugin/cluster-settings.md:389]:
```suggestion
## Node roles that can run externally hosted models
```

File: _ml-commons-plugin/custom-local-models.md (hunk 8 above)
Comment (kolchfa-aws) [reply _ml-commons-plugin/custom-local-models.md:30]: I don't think zip file should be capitalized.

**Pull request merged**